src/compiler/clvm.rs

```diff
@@ -631,9 +631,9 @@ pub fn sha256tree(s: Rc<SExp>) -> Vec<u8> {
             .data()
             .clone()
         }
-        SExp::Nil(_) => sha256tree_from_atom(vec![]),
-        SExp::Integer(_, i) => sha256tree_from_atom(u8_from_number(i.clone())),
-        SExp::QuotedString(_, _, v) => sha256tree_from_atom(v.clone()),
-        SExp::Atom(_, v) => sha256tree_from_atom(v.clone()),
+        SExp::Nil(_) => sha256tree_from_atom(&[]),
+        SExp::Integer(_, i) => sha256tree_from_atom(&u8_from_number(i.clone())),
+        SExp::QuotedString(_, _, v) => sha256tree_from_atom(&v.clone()),
+        SExp::Atom(_, v) => sha256tree_from_atom(&v.clone()),
```

Comment:
it looks like these calls to `clone()` are redundant, since you take a slice now.

Reply:
yes you're right thanks.